recipes/tsil/all/CMakeLists.txt

```diff
@@ -0,0 +1,83 @@
+cmake_minimum_required(VERSION 3.4)
+project(TSIL C)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Release)
+endif()
+
+if(NOT TSIL_SIZE)
+  set(TSIL_SIZE "TSIL_SIZE_LONG") # default
+endif()
+
+if(TSIL_SIZE STREQUAL "TSIL_SIZE_DOUBLE")
+  add_definitions(-DTSIL_SIZE_DOUBLE)
+elseif(TSIL_SIZE STREQUAL "TSIL_SIZE_LONG")
+  add_definitions(-DTSIL_SIZE_LONG)
+else()
+  message(FATAL_ERROR "You must set TSIL_SIZE to either TSIL_SIZE_DOUBLE or TSIL_SIZE_LONG")
+endif()
+
+message(STATUS "Using ${TSIL_SIZE}")
+
+set(SOURCE_DIR source_subfolder)
+
+add_library(tsil
+  ${SOURCE_DIR}/initialize.c
+  ${SOURCE_DIR}/evaluate.c
+  ${SOURCE_DIR}/fevaluate.c
+  ${SOURCE_DIR}/generic.c
+  ${SOURCE_DIR}/rk6.c
+  ${SOURCE_DIR}/rk5.c
+  ${SOURCE_DIR}/functions.c
+  ${SOURCE_DIR}/dilog.c
+  ${SOURCE_DIR}/trilog.c
+  ${SOURCE_DIR}/dBds.c
+  ${SOURCE_DIR}/dSds.c
+  ${SOURCE_DIR}/dTds.c
+  ${SOURCE_DIR}/dUds.c
+  ${SOURCE_DIR}/dsMds.c
+  ${SOURCE_DIR}/initB.c
+  ${SOURCE_DIR}/initS.c
+  ${SOURCE_DIR}/initT.c
+  ${SOURCE_DIR}/initU.c
+  ${SOURCE_DIR}/initV.c
+  ${SOURCE_DIR}/initM.c
+  ${SOURCE_DIR}/special.c
+  ${SOURCE_DIR}/analyticAB.c
+  ${SOURCE_DIR}/analyticI.c
+  ${SOURCE_DIR}/analyticS.c
+  ${SOURCE_DIR}/analyticT.c
+  ${SOURCE_DIR}/analyticTbar.c
+  ${SOURCE_DIR}/analyticU.c
+  ${SOURCE_DIR}/analyticV.c
+  ${SOURCE_DIR}/analyticM.c
+  ${SOURCE_DIR}/setTbar.c
+  ${SOURCE_DIR}/setV.c
+  ${SOURCE_DIR}/setbold.c
+  )
```

Comment:
Can you use glob? Wouldn't it be more generic?

Reply:
I hate to say, but glob is evil: https://www.youtube.com/watch?v=bsXLMQ6WgIk
I used to use glob everywhere too.

Reply:
glob is evil during development, but it doesn't matter in CCI.

Reply:
True, I totally agree.

Reply:
https://gitlab.kitware.com/cmake/cmake/-/merge_requests/1767 using this in production to pull in JSON schemas which are updated automatically...

So far it's been a neutral experience!

Reply:
I could make the following change with glob:
~~~.diff
diff --git a/recipes/tsil/all/CMakeLists.txt b/recipes/tsil/all/CMakeLists.txt
index bc47df02..ed588aa9 100644
--- a/recipes/tsil/all/CMakeLists.txt
+++ b/recipes/tsil/all/CMakeLists.txt
@@ -6,42 +6,19 @@ conan_basic_setup()
 
 set(SOURCE_DIR source_subfolder)
 
-add_library(tsil
-  ${SOURCE_DIR}/initialize.c
-  ${SOURCE_DIR}/evaluate.c
-  ${SOURCE_DIR}/fevaluate.c
-  ${SOURCE_DIR}/generic.c
-  ${SOURCE_DIR}/rk6.c
-  ${SOURCE_DIR}/rk5.c
-  ${SOURCE_DIR}/functions.c
-  ${SOURCE_DIR}/dilog.c
-  ${SOURCE_DIR}/trilog.c
-  ${SOURCE_DIR}/dBds.c
-  ${SOURCE_DIR}/dSds.c
-  ${SOURCE_DIR}/dTds.c
-  ${SOURCE_DIR}/dUds.c
-  ${SOURCE_DIR}/dsMds.c
-  ${SOURCE_DIR}/initB.c
-  ${SOURCE_DIR}/initS.c
-  ${SOURCE_DIR}/initT.c
-  ${SOURCE_DIR}/initU.c
-  ${SOURCE_DIR}/initV.c
-  ${SOURCE_DIR}/initM.c
-  ${SOURCE_DIR}/special.c
-  ${SOURCE_DIR}/analyticAB.c
-  ${SOURCE_DIR}/analyticI.c
-  ${SOURCE_DIR}/analyticS.c
-  ${SOURCE_DIR}/analyticT.c
-  ${SOURCE_DIR}/analyticTbar.c
-  ${SOURCE_DIR}/analyticU.c
-  ${SOURCE_DIR}/analyticV.c
-  ${SOURCE_DIR}/analyticM.c
-  ${SOURCE_DIR}/setTbar.c
-  ${SOURCE_DIR}/setV.c
-  ${SOURCE_DIR}/setbold.c
-  )
-
-add_executable(tsil.x ${SOURCE_DIR}/basecalc.c)
+file(GLOB TSIL_SOURCE "${SOURCE_DIR}/*.c")
+list(FILTER TSIL_SOURCE EXCLUDE REGEX ".*basecalc.c")
+list(FILTER TSIL_SOURCE EXCLUDE REGEX ".*baseST.c")
+list(FILTER TSIL_SOURCE EXCLUDE REGEX ".*baseSTU.c")
+list(FILTER TSIL_SOURCE EXCLUDE REGEX ".*gluinopole.c")
+list(FILTER TSIL_SOURCE EXCLUDE REGEX ".*scalarpole.c")
+list(FILTER TSIL_SOURCE EXCLUDE REGEX ".*test.c")
+list(FILTER TSIL_SOURCE EXCLUDE REGEX ".*test.cpp")
+list(FILTER TSIL_SOURCE EXCLUDE REGEX ".*testprog.c")
+
+add_library(tsil ${TSIL_SOURCE})
+
+add_executable(tsil.x "${SOURCE_DIR}/basecalc.c")
 target_link_libraries(tsil.x tsil m)
 set_target_properties(tsil.x PROPERTIES OUTPUT_NAME "tsil")
 
~~~
I would need to filter out some files that contain a `main`.
What do you think?

Reply:
I doubt this project will change too drastically in the future,

I am happy to keep the fixed list, it can always be changed in the future

Reply:
I see, many filter, so it can break in the other way also, let's keep it with explicit files ;)